provider/defaultingress/ingress.go

```diff
@@ -0,0 +1,269 @@
+			Optional:    true,
+			Validators:  []validator.Map{NotEmptyMapValidator()},
+		},
+		"excluded_namespaces": schema.ListAttribute{
+			Description: "Excluded namespaces for ingress. Format should be a comma-separated list 'value1, value2...'. " +
+				"If no values are specified, all namespaces will be exposed.",
+			ElementType: types.StringType,
+			Optional:    true,
+			Validators: []validator.List{
+				listvalidator.SizeAtLeast(1),
+			},
+		},
+		"route_wildcard_policy": schema.StringAttribute{
+			Description: fmt.Sprintf("Wildcard Policy for ingress. Options are %s. Default is '%s'.",
+				strings.Join(ValidWildcardPolicies, ","), DefaultWildcardPolicy),
+			Optional:   true,
+			Computed:   true,
+			Validators: []validator.String{attrvalidators.EnumValueValidator(ValidWildcardPolicies)},
+		},
+		"route_namespace_ownership_policy": schema.StringAttribute{
+			Description: fmt.Sprintf("Namespace Ownership Policy for ingress. Options are %s. Default is '%s'.",
+				strings.Join(ValidNamespaceOwnershipPolicies, ","), DefaultNamespaceOwnershipPolicy),
+			Optional:   true,
+			Computed:   true,
+			Validators: []validator.String{attrvalidators.EnumValueValidator(ValidNamespaceOwnershipPolicies)},
+		},
+		"cluster_routes_hostname": schema.StringAttribute{
+			Description: "Components route hostname for oauth, console, download.",
+			Optional:    true,
+		},
+		"cluster_routes_tls_secret_ref": schema.StringAttribute{
+			Description: "Components route TLS secret reference for oauth, console, download.",
+			Optional:    true,
+		},
+	}
+}
+
+func PopulateDefaultIngress(ctx context.Context, state *DefaultIngress,
+	client *cmv1.IngressesClient) (*DefaultIngress, error) {
+
+	// in case default ingress is not part of state no need to populate it
+	if state == nil {
+		return nil, nil
+	}
+	ingresses, err := client.List().SendContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, ingress := range ingresses.Items().Slice() {
+		if ingress.Default() {
+			if state == nil {
+				state = &DefaultIngress{}
+			}
+			state.Id = types.StringValue(ingress.ID())
+
+			routeSelectors, ok := ingress.GetRouteSelectors()
+			if ok {
+				state.RouteSelectors, err = common.ConvertStringMapToMapType(routeSelectors)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			excludedNamespaces, ok := ingress.GetExcludedNamespaces()
+			if ok {
+				state.ExcludedNamespaces, err = common.StringArrayToList(excludedNamespaces)
+				if err != nil {
+					return nil, err
+				}
+			}
+			wp, ok := ingress.GetRouteWildcardPolicy()
+			if ok {
+				state.WildcardPolicy = types.StringValue(string(wp))
+			} else {
+				state.WildcardPolicy = types.StringNull()
+			}
+			rnmop, ok := ingress.GetRouteNamespaceOwnershipPolicy()
+			if ok {
+				state.NamespaceOwnershipPolicy = types.StringValue(string(rnmop))
+			} else {
+				state.NamespaceOwnershipPolicy = types.StringNull()
+			}
+			hostname, ok := ingress.GetClusterRoutesHostname()
```

Comment:
Did it before we talked about optional/computed fields, let me fix it

Reply:
Actually i am not sure it will work removing null, how we are going to remove variable in that case? 
Lets say we want to remove hostname so we are removing it and it should be nil how we are gonna get it without setting nil?